Automattic/woocommerce-payments
# fix: translatable strings around currencies list

Fixes #

#### Changes proposed in this Pull Request









I noticed some translation string issues when looking at the `BuildMissingCurrenciesTooltipMessage` and `ListToCommaSeparatedSentencePartConverter` functions in `client/components/currency-information-for-methods/index.js`:
- They're functions returning a string, invoked like a normal function, yet their name is PascalCase like a component
- The `currency` and `currencies` strings might not be directly translatable into the whole sentence. It's better to put the whole sentence for translators to translate (so that they can get a better context)
- The `ListToCommaSeparatedSentencePartConverter` doesn't provide a placeholder for the strings to be translated. The `and` could be taken without context, without considering that some languages might not use the `and` to join list of items, but they might use a `,` comma
- There is a `joinWithConjunction` utility in the codebase, but it's not used anywhere (besides tests). And it doesn't allow for [Oxford commas](https://make.wordpress.org/docs/style-guide/punctuation/commas/)
- The `BuildMissingCurrenciesTooltipMessage` is used inside the `components/currency-information-for-methods` file, but it's also imported directly by other components

So, I made the following changes:
- Moved the `ListToCommaSeparatedSentencePartConverter` to a global scope `client/utils/format-list-of-items.ts` and renamed to something that is closer to [wc_format_list_of_items](https://woocommerce.github.io/code-reference/files/woocommerce-includes-wc-cart-functions.html#function_wc_format_list_of_items)
- Renamed `BuildMissingCurrenciesTooltipMessage` to `getMissingCurrenciesTooltipMessage` and moved it to a separate utility file in the `multi-currency` sub-directory
- Deleted `joinWithConjunction` in favor of `formatListOfItems`
- Moved the `CurrencyInformationForMethods` from `client/components/` (a global directory, made for reuse) to `client/additional-methods-setup/upe-preview-methods-selector/`, since that's the only place it's used

#### Testing instructions





- Ensure you only have USD enabled in your multi-currency store setup
- Ensure you only have card enabled as a payment method
- Go to https://wcpay.test/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fadditional-payment-methods
- Attempt to enable a EUR-based payment method (e.g.: SEPA) and P24
- The tooltip should be displayed below, showing the currencies that will be added to the store
<img width="706" alt="Screenshot 2024-08-14 at 6 19 33 PM" src="https://github.com/user-attachments/assets/9c28b72b-36fc-4e8c-8915-39ce85f90459">


-------------------


**Post merge**
